src/vs/editor/contrib/codeAction/browser/lightBulbWidget.ts

```diff
@@ -173,8 +173,27 @@ export class LightBulbWidget extends Disposable implements IContentWidget {
 		let effectiveLineNumber = lineNumber;
 		let effectiveColumnNumber = 1;
 		if (!lineHasSpace) {
+
+			// Checks if line is empty or starts with any amount of whitespace
+			const isLineEmptyOrIndented = (lineNumber: number): boolean => {
+				const lineContent = model.getLineContent(lineNumber);
+				return /^\s*$|^\s+/.test(lineContent) || lineContent.length <= effectiveColumnNumber;
+			};
+
 			if (lineNumber > 1 && !isFolded(lineNumber - 1)) {
-				effectiveLineNumber -= 1;
+				const lineCount = model.getLineCount();
+				const endLine = lineNumber === lineCount;
+				const prevLineEmptyOrIndented = lineNumber > 1 && isLineEmptyOrIndented(lineNumber - 1);
+				const nextLineEmptyOrIndented = !endLine && isLineEmptyOrIndented(lineNumber + 1);
+				const currLineEmptyOrIndented = isLineEmptyOrIndented(lineNumber);
+				const notEmpty = !nextLineEmptyOrIndented && !prevLineEmptyOrIndented;
+
+				// check above and below. if both are blocked, display lightbulb below.
+				if (prevLineEmptyOrIndented || endLine || (notEmpty && !currLineEmptyOrIndented)) {
+					effectiveLineNumber -= 1;
+				} else if (nextLineEmptyOrIndented || (notEmpty && currLineEmptyOrIndented)) {
```

Comment:
Parsing this, it says "if the next line has whitespace [which is fine], or if the current line has space but neither the above or below line do, then display it on the next line." I'm confused what the that second part is trying to determine 🤔